src/sentry/integrations/slack/utils.py

```diff
@@ -279,27 +285,61 @@ def build_group_attachment(group, event=None, tags=None, identity=None, actions=
     }
 
 
-def build_incident_attachment(incident):
+def build_incident_attachment(incident, trigger=None):
     logo_url = absolute_uri(get_asset_url("sentry", "images/sentry-email-avatar.png"))
-
+    alert_rule = incident.alert_rule
     aggregates = get_incident_aggregates(incident)
 
     if incident.status == IncidentStatus.CLOSED.value:
         status = "Resolved"
         color = RESOLVED_COLOR
+    elif incident.status == IncidentStatus.WARNING.value:
+        status = "Warning"
+        color = LEVEL_TO_COLOR["warning"]
+    elif incident.status == IncidentStatus.CRITICAL.value:
+        status = "Critical"
+        color = LEVEL_TO_COLOR["fatal"]
     else:
+        # Fail safe? Throw error?
         status = "Fired"
         color = LEVEL_TO_COLOR["error"]
 
-    fields = [
-        {"title": "Status", "value": status, "short": True},
-        {"title": "Events", "value": aggregates["count"], "short": True},
-        {"title": "Users", "value": aggregates["unique_users"], "short": True},
-    ]
+    if trigger is None:
+        # Some paths explicitly have a trigger reference, so we send it in with those. Some don't (unfurling), so we do this.
+        # Best guess, grab the most recently updated trigger. It's a pretty good guess with very low failure rate.
+        incident_trigger = (
+            IncidentTrigger.objects.filter(incident=incident).order_by("-date_modified").first()
+        )
+        trigger = incident_trigger.alert_rule_trigger
+    else:
+        incident_trigger = IncidentTrigger.objects.get(
+            incident=incident, alert_rule_trigger=trigger
+        )
+
+    agg_text = QUERY_AGGREGATION_DISPLAY[alert_rule.aggregation]
+
+    agg_value = (
+        aggregates["count"]
+        if alert_rule.aggregation == QueryAggregations.TOTAL.value
+        else aggregates["unique_users"]
+    )
+    is_active = incident_trigger.status == TriggerStatus.ACTIVE
+    is_threshold_type_above = trigger.threshold_type == AlertRuleThresholdType.ABOVE
```

Comment:
`AlertRuleThresholdType.ABOVE.value`

I'd really like to implement an EnumField that automatically converts these into to Python enums so that we don't need to use this everywhere. The django EnumField uses postgres enums, which I don't really want to use since adding a value requires a migration.